src/EventProducerService.cpp

```diff
@@ -23,173 +23,227 @@ void EventProducerService::setController(Controller *cntrl)
 
 void EventProducerService::begin()
 {  
-  if (module_config->currentMode != MODE_UNINITIALISED)
+  if (module_config->currentMode == MODE_UNINITIALISED)
   {
-    setProducedEvents();    
+    uninit = 1;    
   }
 }
 
 void EventProducerService::setProducedEvents()
-{  
-  for (byte i = 0; i < module_config->EE_PRODUCED_EVENTS; i++)
+{ 
+  byte data[4];
+  byte index;
+  for (byte i = 1; i <= module_config->EE_PRODUCED_EVENTS; i++)
   {
-    if (module_config->getEvTableEntry(i) == 0)
+    data[0] = highByte(module_config->nodeNum);
+    data[1] = lowByte(module_config->nodeNum);
+    data[2] = 0;
+    data[3] = i;
+    
+    index = module_config->findExistingEventByEv(1, i);
+    if (index >= module_config->EE_MAX_EVENTS)  //event does not exist so creat default
     {
-      // Event does not exist so create it
-      byte data[4];
-      data[0] = highByte(module_config->nodeNum);
-      data[1] = lowByte(module_config->nodeNum);
-      data[2] = 0;
-      data[3] = i;
-      
-      module_config->writeEvent(i, data);  //index = i
-      module_config->updateEvHashEntry(i);
+      index = module_config->findEventSpace();
     }
+        
+    module_config->writeEvent(index, data);
+    module_config->writeEventEV(index, 1, i);
+    module_config->updateEvHashEntry(index);    
   }    
 }
 
 void EventProducerService::process(UserInterface::RequestedAction requestedAction)
 {
-  // Do this if mode changes to normal
-  if (controller->isSetProdEventTableFlag())
+  // Do this if mode changes from uninitialised to normal
+  if ((uninit) && (module_config->currentMode == MODE_NORMAL))
   {
     setProducedEvents();
-    controller->clearProdEventTableFlag();
+    uninit = 0;
   }
 }
 
-void EventProducerService::sendEvent(bool state, byte index)
+void EventProducerService::createDefault(byte evValue)
```

Comment:
Can we call this function createDefaultProducedEvent() ?
Just to be clear what it is about and what the single evValue is used for?

Reply:
No problem, I'll update.